libcxx/test/std/algorithms/alg.sorting/alg.clamp/ranges.clamp.pass.cpp

```diff
@@ -108,7 +108,33 @@ constexpr bool test() {
     auto prvalue_proj = [](const CheckDoubleMove& x) -> CheckDoubleMove { return x; };
     assert(&std::ranges::clamp(val, low, high, moving_comp, prvalue_proj) == &val);
   }
+  { // Make sure we don't call the projection more than three times per [alg.clamp], see #64717
+    int counter = 0;
+    auto projection_function = [counter](const int value) -> int {
+      assert(counter++ != 3);
```

Comment:
I would just do `++counter` here, no assertion. And then I would check after calling `std::ranges::clamp` that `assert(counter == 3)`.

Reply:
I think you missed this comment!